MagazineLayout/LayoutCore/Types/ElementZIndex.swift

```diff
@@ -0,0 +1,21 @@
+// Created by nate-sentjens on 4/8/19.
+// Copyright © 2019 Airbnb, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Represents the element's position on the z-axis.
+enum ElementZIndex: Int {
+  case background
+  case item
+  case header
+}
```

Comment:
When I was reviewing this, I realized that items will still have undetermined order, since they'll all have zIndex = 1.

Is the ideal behavior for items in a section to have descending z indices? So the first item in a section would have a z index > the last item in a section. I'm thinking it is.

If we agree on that, then these really aren't static values like we've modeled them here. In that case, the logic should actually be something like this (still implemented in `prepare` when we make layout attributes instances):

```swift
let numberOfItems = currentCollectionView.numberOfItems(inSection: sectionIndex)
// ...
// Create header layout attributes if necessary
// ...
newHeaderLayoutAttributes[headerLocation]?.zIndex = numberOfItems + 1
// ...
// Create background layout attributes if necessary
// ...
newBackgroundLayoutAttributes[backgroundLocation]?.zIndex = 0
// ...
// Create item layout attributes if necessary
for itemIndex in 0..<numberOfItems {
// ...
newItemLayoutAttributes[itemLocation]?.zIndex = numberOfItems - itemIndex // range of 1..<numberOfItems
// ...
}
```

LMK what you think! Thanks a bunch of finding this bug and fixing it. I was pumped when I saw the PR email from you :)